src/ng/rootScope.js

```diff
@@ -848,13 +849,14 @@ function $RootScopeProvider() {
 
         clearPhase();
 
-        while (postDigestQueue.length) {
+        while (postDigestQueuePosition < postDigestQueue.length) {
```

Comment:
You are right. Which makes me wonder: Could the same happen with the `asyncQueue` ?